ml-agents-envs/mlagents_envs/base_env.py

```diff
@@ -96,27 +82,29 @@ def __init__(self, obs, reward, done, max_step, agent_id, action_mask):
     def agent_id_to_index(self) -> Dict[AgentId, int]:
         """
         :returns: A Dict that maps agent_id to the index of those agents in
-        this BatchedStepResult.
+        this DecisionSteps.
         """
         if self._agent_id_to_index is None:
             self._agent_id_to_index = {}
             for a_idx, a_id in enumerate(self.agent_id):
                 self._agent_id_to_index[a_id] = a_idx
         return self._agent_id_to_index
 
-    def contains_agent(self, agent_id: AgentId) -> bool:
-        return agent_id in self.agent_id_to_index
+    def __len__(self) -> int:
+        return len(self.agent_id)
+
+    def __contains__(self, item: AgentId) -> bool:
+        return item in self.agent_id_to_index
```

Comment:
You don't actually need this; the default implementation will call `__getitem__`:
https://github.com/python/cpython/blob/d7c567b08f9d7d6aef21b881340a2b72731129db/Lib/_collections_abc.py#L664-L670